src/vcpkg/postbuildlint.cpp

```diff
@@ -840,59 +897,162 @@ namespace vcpkg::PostBuildLint
     struct BuildTypeAndFile
     {
         Path file;
-        BuildType build_type;
+        bool has_static_release = false;
+        bool has_static_debug = false;
+        bool has_dynamic_release = false;
+        bool has_dynamic_debug = false;
     };
 
-    static LintStatus check_crt_linkage_of_libs(const BuildType& expected_build_type,
-                                                const std::vector<Path>& libs,
-                                                const Path& dumpbin_exe)
+    static LocalizedString format_linkage(LinkageType linkage, bool release)
     {
-        auto bad_build_types = std::vector<BuildType>({
-            {ConfigurationType::DEBUG, LinkageType::STATIC},
-            {ConfigurationType::DEBUG, LinkageType::DYNAMIC},
-            {ConfigurationType::RELEASE, LinkageType::STATIC},
-            {ConfigurationType::RELEASE, LinkageType::DYNAMIC},
-        });
-        Util::erase_remove(bad_build_types, expected_build_type);
+        switch (linkage)
+        {
+            case LinkageType::DYNAMIC:
+                if (release)
+                {
+                    return msg::format(msgLinkageDynamicRelease);
+                }
+                else
+                {
+                    return msg::format(msgLinkageDynamicDebug);
+                }
+                break;
+            case LinkageType::STATIC:
+                if (release)
+                {
+                    return msg::format(msgLinkageStaticRelease);
+                }
+                else
+                {
+                    return msg::format(msgLinkageStaticRelease);
+                }
+                break;
+            default: Checks::unreachable(VCPKG_LINE_INFO);
+        }
+    }
 
+    static LintStatus check_crt_linkage_of_libs(const Filesystem& fs,
+                                                const BuildInfo& build_info,
+                                                bool expect_release,
+                                                const std::vector<Path>& libs)
+    {
+        (void)build_info;
+        (void)expect_release;
         std::vector<BuildTypeAndFile> libs_with_invalid_crt;
-
         for (const Path& lib : libs)
         {
-            auto cmd_line = Command(dumpbin_exe).string_arg("/directives").string_arg(lib);
-            const auto maybe_output = flatten_out(cmd_execute_and_capture_output(cmd_line), "dumpbin");
-            if (const auto output = maybe_output.get())
+            auto maybe_lib_info = fs.try_open_for_read(lib).then(
+                [](ReadFilePointer&& lib_file) { return read_lib_information(lib_file); });
+
+            if (!maybe_lib_info.has_value())
+            {
+                continue;
+            }
+
+            auto&& lib_info = maybe_lib_info.value_or_exit(VCPKG_LINE_INFO);
+            Debug::println("The lib " + lib.native() +
+                           " has directives: " + Strings::join(" ", lib_info.linker_directives));
```

Comment:
```suggestion
            Debug::println("The lib ", lib.native(),
                           " has directives: ", Strings::join(" ", lib_info.linker_directives));
```

Reply:
I don't think this is actually better because it explodes code size to do the exact same thing. (The concat-like interfaces can do useful things when they are converting non-strings into strings as part of the same operation but that's not happening here)

Reply:
OTOH the concatenation operations are pushed inside the conditional instead of running always.

Reply:
>OTOH the concatenation operations are pushed inside the conditional instead of running always. 

That's fair, OK.

>Shouldn't the code size end up approximately the same?

No, because as written this ends up (effectively) calling the same `operator+(std::string&&, const std::string&)` N times which is shared everywhere else in the codebase, while the concat form forces a duplication of all the concat code for every possible combination of parameters to this `Debug::` call.